### PR TITLE
Further simplify the plugin installation process.

### PR DESCRIPTION
**Pull request by DennisHeimbigner**
re: https://github.com/Unidata/netcdf-c/pull/2431

Hat tip to [DWesl](https://github.com/DWesl) and [opoplawski](https://github.com/opoplawski) for their help.

With some help, I found out how to get rid of the tmp_LTLIBARIES
hack and replace it with check_LTLIBRARIES.

Using check_LTLIBRARIES means also that the test libraries are not
built until "make check" is executed.

This PR replaces the one above.

### CODE REVIEW

File: plugins/Makefile.am [hunk 1]
```diff
@@ -4,31 +4,28 @@
...
-
-# This is where the plugins are to be installed
-if ENABLE_PLUGIN_DIR
+#if ENABLE_PLUGIN_DIR
 plugindir = @PLUGIN_INSTALL_DIR@
-else
-plugindir = ${ALTPLUGINDIR}
-endif
+#else
+plugindir = ${prefix}
```
Comment (DWesl) [inline plugins/Makefile.am:10]: There's nothing tying `--enable-plugins` to `--with-plugin-dir`; I think just specifying `--enable-plugins` without `--with-plugin-dir` will put the plugins into `${prefix}`.  Would `${libexecdir}/netcdf-c` or similar make more sense for the fallback location, would a better fallback be not installing those plugins at all (as implied by the old value), or is there some reasoning for this choice I'm not aware of?

File: plugins/Makefile.am [hunk 2]
```diff
@@ -72,20 +70,21 @@ if ENABLE_PLUGINS
 
 # The NCZarr codec libraries
 lib__nczstdfilters_la_SOURCES = NCZstdfilters.c
+lib__nczstdfilters_la_LDFLAGS = ${INSTALLFLAGS}
 lib__nczhdf5filters_la_SOURCES = NCZhdf5filters.c
-
+lib__nczhdf5filters_la_LDFLAGS = ${INSTALLFLAGS}
```
Comment (DWesl) [inline plugins/Makefile.am:75]: You seem to have commented out the `INSTALLFLAGS` assignment on line 20; should the `_LDFLAGS` lines with just that be deleted (allowing the default `AM_LDFLAGS` to serve those plugins, or the `INSTALLFLAGS` assignment uncommented?

**Review by DWesl — COMMENTED** [review]
Would it be worth adding a step checking `make DESTDIR=/tmp/pretend-root install` doesn't install any of the test plugins to one of the CI jobs?  
`if ! find /tmp/pretend-root -name lib__nctestplugin.la; exit 1`
might work as the condition.

File: plugins/Makefile.am (hunk 1 above)
Comment (DWesl) [reply plugins/Makefile.am]: New version is basically what I ended up with.

File: plugins/Makefile.am (hunk 2 above)
Comment (DWesl) [reply plugins/Makefile.am:81]: Uncommented the `INSTALLFLAGS` definition and started it with `AM_LDFLAGS`, that'll help Cygwin builds (and probably Windows).

Doesn't look like there's a "resolve" button like I've seen on some of these.

**Comment by opoplawski** [comment]
This seems to work for me.  Thanks.

**Comment by WardF** [comment]
@DennisHeimbigner some conflicts have crept in, and Github is (again) not letting me resolve them.  I'll try to sort that out, but are you able to make changes and/or have a moment to address them?

**Comment by DennisHeimbigner** [comment]
Ok, let me see what I can do.

**Comment by DennisHeimbigner** [comment]
This PR was apparently made moot by other, later PRs.

**Pull request closed without merging**